src/python/pants/notes/2.0.x.rst

```diff
@@ -6,6 +6,26 @@ This document describes releases leading up to the ``2.0.x`` ``stable`` series.
 See https://www.pantsbuild.org/v2.0/docs/release-notes-2-0 for an overview of the changes in this release, and
 https://www.pantsbuild.org/docs/plugin-upgrade-guide for a plugin upgrade guide.
 
+2.0.1rc4 (12/09/2020)
+---------------------
+
+Bugfixes
+~~~~~~~~
+
+* Revert "Move graph cycle detection to rust. (#11202)" (cherrypick of #11272) (#11277)
+  `PR #11202 <https://github.com/pantsbuild/pants/pull/11202>`_
+  `PR #11277 <https://github.com/pantsbuild/pants/pull/11277>`_
+
+* Increase Pants' python recursion limit by default, and allow it to be overridden. (cherrypick of #11271) (#11276)
+  `PR #11276 <https://github.com/pantsbuild/pants/pull/11276>`_
+
+Internal
```

Comment:
This should be left off

Reply:
Not sure I agree. Including all commits here makes sense to me, as this is our complete changelog (and that's what the docs say)... the per-version overviews (https://www.pantsbuild.org/docs/versions) are the summaries.

Reply:
Hm we made the decision about a month ago to leave off internal only changes, under the logic that the changelog's explicit purpose is to help users see what's changed. Git log is used for the complete history if that's what you want. https://github.com/pantsbuild/pants/pull/10916

There was a Slack convo too, and Benjy's new PR https://github.com/pantsbuild/pants/pull/11280 builds off of this reasoning.

Reply:
> There was a Slack convo too, and Benjy's new PR #11280 builds off of this reasoning.

Ok. That didn't end up in [the docs](https://www.pantsbuild.org/docs/release-process), but having it in the script would be great. I think that once it is in the script we can begin applying it... so probably not existing stable branches, for consistency.

Reply:
It's in the script in master, but we didn't cherry pick. Iiuc, I updated the 2.2 docs but not 2.1.

We've been applying this reasoning in 1.30.2 and 2.0 and 2.1 docs for the past month, so I'm not sure consistency would be very relevant for these two new releases.

Reply:
It's clear what we're going to do moving forward, but I don't think that it's worth another CI burn here.